src/lib/profiles/data-management/Current/SubscriptionEngine.cpp

```diff
@@ -1515,7 +1520,675 @@ void SubscriptionEngine::OnCustomCommand(nl::Weave::ExchangeContext * aEC, const
+                (elementTracker->mStatusCode == Weave::Profiles::Common::kStatus_AccessDenied))
+            {
+                lVLBuilder.AddNull();
+            }
+            else if ((elementTracker->mProfileId == Weave::Profiles::kWeaveProfile_WDM) &&
+                     (elementTracker->mStatusCode == kStatus_InvalidPath))
+            {
+                lVLBuilder.AddNull();
+            }
+            else if (context->mpCatalog->Locate(elementTracker->mTraitDataHandle, &dataSource) == WEAVE_NO_ERROR)
+            {
+                lVLBuilder.AddVersion(dataSource->GetVersion());
+            }
+            else
+            {
+                lVLBuilder.AddNull();
+            }
+            elementTracker++;
+        }
+        lVLBuilder.EndOfVersionList();
+        SuccessOrExit(lVLBuilder.GetError());
+    }
+
+    elementTracker = static_cast<struct StatusDataHandleElement *>(context->mpFirstStatusDataHandleElement);
+    {
+        StatusList::Builder & lSLBuilder = updateResponseBuilder.CreateStatusListBuilder();
+        for (uint32_t j = 0; j < context->mNumDataElements; j++)
+        {
+            lSLBuilder.AddStatus(elementTracker->mProfileId, elementTracker->mStatusCode);
+            elementTracker++;
+        }
+        lSLBuilder.EndOfStatusList();
+        SuccessOrExit(lSLBuilder.GetError());
+    }
+
+    updateResponseBuilder.EndOfResponse();
+    SuccessOrExit(updateResponseBuilder.GetError());
+
+    aWriter.Finalize();
+
+    WeaveLogDetail(DataManagement, "ConstructStatusListVersionList success with number of elements %d", context->mNumDataElements);
+
+exit:
+    if (err != WEAVE_NO_ERROR)
+    {
+        aWriter = checkpoint;
+    }
+}
+
+void SubscriptionEngine::BuildStatusDataHandleElement(PacketBuffer * pBuf, TraitDataHandle aTraitDataHandle, WEAVE_ERROR & err,
+                                                      uint32_t aCurrentIndex)
+{
+    uint32_t profileId  = 0;
+    uint16_t statusCode = 0;
+
+    if (WEAVE_ERROR_ACCESS_DENIED == err)
+    {
+        profileId  = Weave::Profiles::kWeaveProfile_Common;
+        statusCode = Weave::Profiles::Common::kStatus_AccessDenied;
+        err        = WEAVE_NO_ERROR;
+    }
+    else if (WEAVE_ERROR_INVALID_PROFILE_ID == err)
+    {
+        profileId  = Weave::Profiles::kWeaveProfile_WDM;
+        statusCode = kStatus_InvalidPath;
+        err        = WEAVE_NO_ERROR;
+    }
+    else if (WEAVE_ERROR_WDM_VERSION_MISMATCH == err)
+    {
+        profileId  = Weave::Profiles::kWeaveProfile_WDM;
+        statusCode = kStatus_VersionMismatch;
+        err        = WEAVE_NO_ERROR;
+    }
+    else if (WEAVE_NO_ERROR == err)
+    {
+        profileId  = Weave::Profiles::kWeaveProfile_Common;
+        statusCode = Weave::Profiles::Common::kStatus_Success;
+    }
+    else
+    {
+        profileId  = Weave::Profiles::kWeaveProfile_Common;
+        statusCode = Weave::Profiles::Common::kStatus_InternalError;
```

Comment:
Log a detail message just in case we need to have some additional intelligence to create a more specific status code.  Log message should include the err code we're dispatching on as well as which update element this internal error pertained to.

Reply:
Done, thanks